### ISSUE TITLE
[Suggestion] Expand codebase

### ISSUE THREAD

**Issue by cloudrex** [imba/imba#193]
Suggestion: The codebase is very dense, specially in certain files such as `nodes.imba`. Expand the codebase, and separate responsibility of the functionality within these files.

Why?

1. Organization
2. Spaghetti code prevention
3. Easy reference
4. Better for new contributors
5. Huge files cause problems in certain editors

**Comment by athif23** [comment]
I agree, but the problem is the author @somebee is busy. We can only expect someone who have time and mind to do that. I did have tried to refactor them, but I'm afraid my code have different taste to others. So, I think it would be good if we have or make some code guideline, to make sure the codes are consistent in all files. 

Or do you have a suggestion in how we can do it?

**Comment by cloudrex** [comment]
Personally, I'd approach this in the good ol' way--move specific related chunks of code into their own corresponding files, splitting the codebase amount different files and folders.

**Comment by pushqrdx** [comment]
i'd add, doing this without turning it into a huge nested file structure though, because node.imba alone if split incorrectly is enough to scare people away 😂

**Comment by cloudrex** [comment]
> i'd add, doing this without turning it into a huge nested file structure though, because node.imba alone if split incorrectly is enough to scare people away 😂

Yea that sounds logical. It could be split onto same-level directories. That's how I proceed with most of my project structures.
